config.go

```diff
@@ -21,14 +26,51 @@ type IPAMConfig struct {
 
 // Net loads the options of the CNI network configuration file
 type Net struct {
-	Name string      `json:"name"`
-	IPAM *IPAMConfig `json:"ipam"`
+	Name     string      `json:"name"`
+	IPAM     *IPAMConfig `json:"ipam"`
+	BrSubnet string      `json:"bridgeSubnet"`
+	BrName   string      `json:"bridge"`
 }
 
-// LoadIPAMConfig loads the IPAM configuration from the given bytes
-func LoadIPAMConfig(bytes []byte, args string) (*IPAMConfig, error) {
-	n := Net{}
-	if err := json.Unmarshal(bytes, &n); err != nil {
+func (n *Net) getSubnetSize() (prefixSize string) {
+	if n.IPAM.SubnetPrefixSize != "" {
+		prefixSize = n.IPAM.SubnetPrefixSize
+	} else {
+		prefixSize = "/" + strings.SplitN(n.BrSubnet, "/", 2)[1]
+	}
+	return prefixSize
+}
+
+func (n *Net) getMetadataRoute() (route types.Route, err error) {
+	metadataAddress := os.Getenv("RANCHER_METADATA_ADDRESS")
+	if metadataAddress == "" {
+		metadataAddress = metadata.DefaultMetadataAddress
+	}
+	_, metadataNet, err := net.ParseCIDR(fmt.Sprintf("%s/32", metadataAddress))
+	if err != nil {
+		return route, err
+	}
+
+	l, err := netlink.LinkByName(n.BrName)
+	if err != nil {
+		return route, err
+	}
+	addrs, err := netlink.AddrList(l, netlink.FAMILY_V4)
+	if err != nil {
+		return route, err
+	}
+	if len(addrs) > 1 || len(addrs) == 0 {
```

Comment:
It would be nice to split this into two separate conditions and add detailed error messages. Would be helpful in debugging when there is a problem.